### PR TITLE
test/fio: extend fio objectstore plugin to better simulate OSD behavior

### PR DESCRIPTION
**Pull request by ifed01**
Signed-off-by: Igor Fedotov <ifedotov@suse.com>

### CODE REVIEW

**Comment by ifed01** [comment]
@cbodley - ping

File: src/test/fio/fio_ceph_objectstore.cc [hunk 1]
```diff
@@ -54,19 +68,156 @@ static std::vector<fio_option> ceph_options{
...
 };
 
 
+struct Collection {
+  spg_t pg;
+  coll_t cid;
+  ObjectStore::Sequencer sequencer;
+  // Can't use mutex directly in vectors hence dynamic allocation
+
+  ceph::shared_ptr<std::mutex> lock;
```
Comment (cbodley) [inline src/test/fio/fio_ceph_objectstore.cc]: `std::unique_ptr` will also work here (`vector` only requires it to be movable *or* copyable), and would save an extra allocation on construction

**Comment by ifed01** [comment]
@cbodley - fixed

**Comment by ifed01** [comment]
retest this please

**Pull request merged**